src/Handler/IteratorHandler.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Handler;
+
+use JMS\Serializer\DeserializationContext;
+use JMS\Serializer\GraphNavigatorInterface;
+use JMS\Serializer\SerializationContext;
+use JMS\Serializer\Visitor\DeserializationVisitorInterface;
+use JMS\Serializer\Visitor\SerializationVisitorInterface;
+
+final class IteratorHandler implements SubscribingHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribingMethods()
+    {
+        $methods = [];
+        $formats = ['json', 'xml', 'yml'];
+        $iteratorTypes = [
+            \Iterator::class,
+            \ArrayIterator::class,
+        ];
+
+        foreach ($iteratorTypes as $type) {
+            foreach ($formats as $format) {
+                $methods[] = [
+                    'direction' => GraphNavigatorInterface::DIRECTION_SERIALIZATION,
+                    'type' => $type,
+                    'format' => $format,
+                    'method' => 'serializeIterator',
+                ];
+
+                $methods[] = [
+                    'direction' => GraphNavigatorInterface::DIRECTION_DESERIALIZATION,
+                    'type' => $type,
+                    'format' => $format,
+                    'method' => 'deserializeIterator',
+                ];
+            }
+        }
+
+        foreach ($formats as $format) {
+            $methods[] = [
+                'direction' => GraphNavigatorInterface::DIRECTION_SERIALIZATION,
+                'type' => \Generator::class,
+                'format' => $format,
+                'method' => 'serializeIterator',
+            ];
+
+            $methods[] = [
+                'direction' => GraphNavigatorInterface::DIRECTION_DESERIALIZATION,
+                'type' => \Generator::class,
+                'format' => $format,
+                'method' => 'deserializeGenerator',
+            ];
+        }
+
+        return $methods;
+    }
+
+    /**
+     * @return array|\ArrayObject
+     */
+    public function serializeIterator(
+        SerializationVisitorInterface $visitor,
+        \Iterator $iterator,
+        array $type,
+        SerializationContext $context
+    ) {
+        $type['name'] = 'array';
+
+        $context->stopVisiting($iterator);
+        $result = $visitor->visitArray(iterator_to_array($iterator), $type);
+        $context->startVisiting($iterator);
+
+        return $result;
+    }
+
+    /**
+     * @param mixed $data
+     */
+    public function deserializeIterator(
+        DeserializationVisitorInterface $visitor,
+        $data,
+        array $type,
+        DeserializationContext $context
+    ): \Iterator {
+        $type['name'] = 'array';
+
+        return new \ArrayIterator($visitor->visitArray($data, $type));
```

Comment:
I wonder whether it would make sense and also whether it's possible to make this dynamic for every type that implements `Iterator` except `Generator`? It seems there's [a lot SPL Iterators](http://php.net/manual/en/spl.iterators.php) and also might then work for custom ones.

Reply:
It is possible to serialise any Iterator, but the problem is with deserialising - I see no way to deserialise data into any Iterators that modify the behaviour  - like NoRewind / Limit / Filter / Regex Iterators. It might be hard to deserialise Iterators that depends on Filesystem. 
I'll take a look into RecursiveArrayIterator as it might be easy to add it.

Reply:
Kk, just asked :) never mind